src/main/java/org/opensearch/knn/common/KNNConstants.java

```diff
@@ -73,6 +73,7 @@ public class KNNConstants {
     public static final VectorDataType DEFAULT_VECTOR_DATA_TYPE_FIELD = VectorDataType.FLOAT;
 
     public static final String RADIAL_SEARCH_KEY = "radial_search";
+    public static final String MODEL_VERSION = "model_version";
```

Comment:
What is the purpose of this model version ?

Reply:
It helps prevent null pointer exceptions, since we can tell which variables are present in the metadata based on the version.  Additionally, it is required for re-scoring features that are being added as part of disk-based search.